demo/admin/src/products/ProductForm.tsx

```diff
@@ -62,10 +64,18 @@ const rootBlocks = {
     image: DamImageBlock,
 };
 
-type FormValues = Omit<GQLProductFormManualFragment, "price" | "image"> & {
-    price: string;
-    image: BlockState<typeof rootBlocks.image>;
-};
+type FormValues = MergeDeep<
+    GQLProductFormManualFragment,
+    {
+        price: string;
+        image: BlockState<typeof rootBlocks.image>;
+        packageDimensions: {
+            width: string;
+            height: string;
+            depth: string;
+        };
+    }
+>;
```

Comment:
That is very special, uses MergeDeep from a new "type-fest" dependency. Problem is that `{ foo: string } &  { foo: number }` results in `{ foo: string | number }`, while `Merge` results in `{ foo: number }`. And `MergeDeep` does this also for nested objects.

type-fest is a [very popular](https://npmtrends.com/ts-essentials-vs-ts-toolbelt-vs-type-fest-vs-type-zoo-vs-typelevel-ts-vs-typepark-vs-typical-vs-typical-ts-vs-utility-types) library

Reply:
Do we "need"? Shouldn't this be also achievable using the  (now removed) Omit statement? 🤔

Reply:
Manual Omit is possible, but very verbose and complicated for nested objects. That's why I was looking for an alternative.

```
Omit<GQLProductFormManualFragment, "price" | "image" | "packageDimensions"> & {
    price: string;
    image: BlockState<typeof rootBlocks.image>;
    packageDimensions: Omit<GQLProductFormManualFragment["packageDimensions"], "width" | "height" | "depth"> & {
        width: string;
        height: string;
        depth: string;
    }
}
```

(as packageDimensions currently has only those thre fields this could be simplified, but let's assume there are other eg string fields in packageDimensions)